From 2408b3b59a81b4ea0ffcd080b3cb05c70fd024fa Mon Sep 17 00:00:00 2001
From: Zeng Heng <zengheng4@huawei.com>
Date: Thu, 29 May 2025 09:34:50 +0800
Subject: [PATCH 722/960] downstream: fs/resctrl: Restore default settings for
 all resctrl_res_level

Each time a new control group is created, all the resctrl_res_level items
need to be initialized with default values to prevent the partid from
retaining old configurations due to previous usage.

Signed-off-by: Zeng Heng <zengheng4@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 fs/resctrl/rdtgroup.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/fs/resctrl/rdtgroup.c b/fs/resctrl/rdtgroup.c
index 9252781360c4..a42da66ae88e 100644
--- a/fs/resctrl/rdtgroup.c
+++ b/fs/resctrl/rdtgroup.c
@@ -3166,7 +3166,7 @@ static int rdtgroup_init_cat(struct resctrl_schema *s, u32 closid)
 }
 
 /* Initialize MBA resource with default values. */
-static void rdtgroup_init_mba(struct rdt_resource *r, u32 closid)
+static void rdtgroup_init_res(struct rdt_resource *r, u32 closid)
 {
 	struct resctrl_staged_config *cfg;
 	struct rdt_domain *d;
@@ -3194,15 +3194,16 @@ static int rdtgroup_init_alloc(struct rdtgroup *rdtgrp)
 
 	list_for_each_entry(s, &resctrl_schema_all, list) {
 		r = s->res;
-		if (r->rid == RDT_RESOURCE_MBA ||
-		    r->rid == RDT_RESOURCE_SMBA) {
-			rdtgroup_init_mba(r, rdtgrp->closid);
-			if (is_mba_sc(r))
-				continue;
-		} else {
+		if (r->rid == RDT_RESOURCE_L2 ||
+		    r->rid == RDT_RESOURCE_L3) {
 			ret = rdtgroup_init_cat(s, rdtgrp->closid);
 			if (ret < 0)
 				goto out;
+
+		} else {
+			rdtgroup_init_res(r, rdtgrp->closid);
+			if (is_mba_sc(r))
+				continue;
 		}
 
 		ret = resctrl_arch_update_domains(r, rdtgrp->closid);
-- 
2.43.0

